pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -117,175 +121,151 @@ func labelMachineSetNodes(client runtimeclient.Client, ms *mapiv1beta1.MachineSe
-					glog.Errorf("error querying api for machineAutoscaler object: %v, retrying...", err)
-					return false, nil
-				}
-				return true, nil
-			})
-			glog.Info("Deleted machineAutoscaler object")
-
-			wait.PollImmediate(1*time.Second, e2e.WaitShort, func() (bool, error) {
-				if err := client.Delete(context.TODO(), &clusterAutoscaler); err != nil {
-					glog.Errorf("error querying api for clusterAutoscaler object: %v, retrying...", err)
-					return false, nil
-				}
-				return true, nil
-			})
-			glog.Info("Deleted clusterAutoscaler object")
-		}()
-
-		nodeTestLabel := fmt.Sprintf("machine.openshift.io/autoscaling-test-%v", string(uuid.NewUUID()))
+		g.By("Creating MachineAutoscaler objects")
+		machineAutoscaler := machineAutoscalerResource(&targetMachineSet, 1, initialNumberOfReplicas+1)
+		err = client.Create(context.TODO(), machineAutoscaler)
+		o.Expect(err).NotTo(o.HaveOccurred())
 
-		// Label all nodes belonging to the machineset (before scale up phase)
+		g.By("Labeling all nodes belonging to the machineset (before scale up phase)")
 		err = labelMachineSetNodes(client, &targetMachineSet, nodeTestLabel)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		glog.Info("Get nodeList")
 		nodeList := corev1.NodeList{}
-		err = wait.PollImmediate(1*time.Second, e2e.WaitMedium, func() (bool, error) {
-			if err := client.List(context.TODO(), runtimeclient.MatchingLabels(map[string]string{nodeTestLabel: ""}), &nodeList); err != nil {
-				glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
-				return false, err
-			}
-			return true, nil
-		})
+		err = client.List(context.TODO(), runtimeclient.MatchingLabels(map[string]string{nodeTestLabel: ""}), &nodeList)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		nodeGroupInitialTotalNodes := len(nodeList.Items)
 		glog.Infof("Cluster initial number of nodes in node group %v is %d", targetMachineSet.Name, nodeGroupInitialTotalNodes)
 
-		glog.Info("Create workload")
-
-		err = wait.PollImmediate(1*time.Second, e2e.WaitMedium, func() (bool, error) {
-			if err := client.Create(context.TODO(), workLoad); err != nil {
-				glog.Errorf("error querying api for workLoad object: %v, retrying...", err)
-				return false, err
-			}
-			return true, nil
-		})
+		g.By("Creating workload")
+		err = client.Create(context.TODO(), newWorkLoad())
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		glog.Info("Waiting for cluster to scale out number of replicas")
+		g.By("Waiting for cluster to scale out number of replicas")
 		err = wait.PollImmediate(5*time.Second, e2e.WaitLong, func() (bool, error) {
-			msKey := types.NamespacedName{
-				Namespace: e2e.TestContext.MachineApiNamespace,
-				Name:      targetMachineSet.Name,
-			}
 			ms := &mapiv1beta1.MachineSet{}
-			if err := client.Get(context.TODO(), msKey, ms); err != nil {
-				glog.Errorf("error querying api for clusterAutoscaler object: %v, retrying...", err)
+			if err := client.Get(context.TODO(), targetMachineSetKey, ms); err != nil {
```

Comment:
can we create `getMachineSet` in utils?

Reply:
I removed the function on purpose since it can be replaced by the one-liner you are commenting onto.

Reply:
Done